src/interfaces.ts

```diff
@@ -123,10 +90,21 @@ export interface Connection {
   contract: string;
   list: () => Promise<TableMetadata[]>;
   create: (
-    query: string,
-    options?: CreateTableOptions
+    /** The schema that defines the columns and constraints of the table,
+     *  e.g.
+     `id int NOT NULL,
+      name char(50) NOT NULL,
+      favorite_food char(50),
+      PRIMARY KEY (id)`
+     */
+    schema: string,
+    /** an optional prefix to the tablename that will be assigned to this table.
+     *  If supplied, it must conform to the rules of SQL table names
+     **/
+    prefix?: string
   ) => Promise<ContractReceipt>;
-  query: (query: string) => Promise<null | ReadQueryResult>;
+  read: (query: string) => Promise<ReadQueryResult>;
+  write: (query: string) => Promise<WriteQueryResult>;
   hash: (query: string) => Promise<StructureHashReceipt>;
-  receipt: (txnHash: string) => Promise<ReceiptResult>;
+  receipt: (txnHash: string) => Promise<ReceiptResult | void>;
```

Comment:
If the transaction hasn't been digested return undefined

Reply:
Void should be reserved for functions that don't return _anything_, rather than sometimes return undefined: https://www.typescripttutorial.net/typescript-tutorial/typescript-void-type/

```suggestion
  receipt: (txnHash: string) => Promise<ReceiptResult | undefined>;
```